File: holiday_warmups/removeEnds.rb


def remove_ends (str_input) 
    counter = 1
    str_output = []
    if (str_input.length < 3) 
        return str_output.join()
    else  
        while counter.to_i<str_input.length-1 
            str_output.push (str_input[counter])
            counter += 1;
        end
        return str_output.join('')
    end
end

p(remove_ends("SEI Rocks!"));
p(remove_ends("a"));
p(remove_ends("This is a test"));
p(remove_ends("Fin"));

File: holiday_warmups/removeEnds.js
const removeEnds = (strInput) => {
  strArrayInput = [];
  strArrayOutput = [];
  strArrayInput = strInput.split("");

  if (strArrayInput.length < 3) {
    strArrayOutput = [""];
    return strArrayOutput.join("");
  } else {
    for (let i = 1; i < strArrayInput.length - 1; i++) {
      strArrayOutput.push(strArrayInput[i]);
    }
    return strArrayOutput.join("");
  }
};

console.log(removeEnds("SEI Rocks!"));
console.log(removeEnds("a"));
console.log(removeEnds("This is a test"));
console.log(removeEnds("Fin"));
